corporate/lib/stripe.py

```diff
@@ -3244,6 +3245,28 @@ def sync_license_ledger_if_needed(self) -> None:
             if plan is None:
                 return
 
+    def get_push_service_validity_dict(self) -> RemoteRealmDictValue:
+        customer = self.get_customer()
+        if customer is None:
+            return {"can_push": True, "expected_end_timestamp": None}
+
+        current_plan = get_current_plan_by_customer(customer)
+        if current_plan is None:
+            return {"can_push": True, "expected_end_timestamp": None}
+
+        expected_end_timestamp = None
+        if current_plan.status in [
+            CustomerPlan.DOWNGRADE_AT_END_OF_CYCLE,
+            CustomerPlan.DOWNGRADE_AT_END_OF_FREE_TRIAL,
+        ]:
+            expected_end_timestamp = datetime_to_timestamp(
+                self.get_next_billing_cycle(current_plan)
+            )
+        return {
+            "can_push": True,
+            "expected_end_timestamp": expected_end_timestamp,
+        }
```

Comment:
I think we don't need to think about that for now, but the idea would be to cache the answers on `RemoteRealm`/`RemoteZulipServer`. Not trivial because of the time variable, so we can not worry about that for now.